matsim/src/main/java/org/matsim/core/mobsim/jdeqsim/JDEQSimulation.java

```diff
@@ -79,17 +82,15 @@ public void run() {
 	}
 
 	protected void initializeRoads() {
-		HashMap<Id<Link>, Road> allRoads = new HashMap<>();
 		for (Link link : scenario.getNetwork().getLinks().values()) {
-			allRoads.put(link.getId(), new Road(scheduler, link));
+			allRoads.put(link.getId(), new Road(scheduler, link, config));
 		}
-		Road.setAllRoads(allRoads);
 	}
 
 	protected void initializeVehicles() {
 		for (Person person : scenario.getPopulation().getPersons().values()) {
 			// the vehicle registers itself to the scheduler
-			new Vehicle(scheduler, person, activityDurationInterpretation);
+			new Vehicle(scheduler, person, activityDurationInterpretation, allRoads, messageFactory);
```

Comment:
These Vehicles need allRoads and messageFactory to work. They can take these objects from static fields or be provided with them at creation time (or maybe at simulation time). Both approaches are fine. But if we use static fields then we can run only a single phys simulation at a time. Because we have only one instance of allRoads in this case. All the others will just overwrite that static field with a new value.
If we run tests in parallel then multiple phys simulations happen at the same time so we have to not store that allRoads as a static field.